src/components/diagrams/diagram-pane.js

```diff
@@ -464,45 +466,205 @@ export function DiagramPane({
         // Note: studyUuid, and loadNetwork don't change
     }, [
         studyUpdatedForce,
-        dispatch,
         studyUuid,
         updateDiagram,
         closeDiagramViews,
         network,
     ]);
 
     /**
-     * DIAGRAM SIZE COMPUTATION
+     * DIAGRAM SIZE CALCULATION
      */
 
-    const [computedHeight, setComputedHeight] = useState();
+    // This function is called by the diagram's contents, when they get their sizes from the backend.
+    const setDiagramSize = (diagramId, diagramType, width, height) => {
+        // Let's update the stored values if they are new
+        const storedValues = diagramContentSizes?.get(diagramType + diagramId);
```

Comment:
Those values are not deleted when the diagram is closed nope ?

Reply:
No they are not.
This way, if the diagram is closed and opened again, the animation is smoother because the diagram's values are already there.
Also, because it is not needed to delete them, I chose to let them as is so there is less clutter in the code.